automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/api/orch/OrchestrationApiSpec.scala

```diff
@@ -0,0 +1,130 @@
+package org.broadinstitute.dsde.firecloud.test.api.orch
+
+import com.google.api.client.googleapis.json.GoogleJsonResponseException
+import com.google.api.services.bigquery.model.GetQueryResultsResponse
+import org.broadinstitute.dsde.firecloud.api.{APIException, Orchestration, Rawls}
+import org.broadinstitute.dsde.firecloud.auth.AuthToken
+import org.broadinstitute.dsde.firecloud.config.{Credentials, UserPool}
+import org.broadinstitute.dsde.firecloud.fixture.BillingFixtures
+import org.scalatest.concurrent.{Eventually, ScalaFutures}
+import org.scalatest.time.{Seconds, Span}
+import org.scalatest.{FreeSpec, Matchers}
+import org.broadinstitute.dsde.firecloud.dao.Google.googleBigQueryDAO
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+
+class OrchestrationApiSpec extends FreeSpec with Matchers with ScalaFutures with Eventually
+  with BillingFixtures {
+  implicit override val patienceConfig: PatienceConfig = PatienceConfig(timeout = scaled(Span(5, Seconds)))
+
+  "Orchestration" - {
+    "should grant and remove google role access" in {
+      val ownerUser: Credentials = UserPool.chooseProjectOwner
+      val ownerToken: AuthToken = ownerUser.makeAuthToken()
+      val role = "bigquery.jobUser"
+
+      val user: Credentials = UserPool.chooseStudent
+      val userToken: AuthToken = user.makeAuthToken()
+
+      // Willy Shakes uses this insult twice
+
+      val shakespeareQuery = "SELECT COUNT(*) AS scullion_count FROM publicdata.samples.shakespeare WHERE word='scullion'"
+
+      def assertExpectedShakespeareResult(response: GetQueryResultsResponse): Unit = {
+        val resultRows = response.getRows
+        resultRows.size shouldBe 1
+        val resultFields = resultRows.get(0).getF
+        resultFields.size shouldBe 1
+        resultFields.get(0).getV.toString shouldBe "2"
+      }
+
+      withBillingProject("auto-goog-role") { projectName =>
+        val accessDeniedMessage = s"The user ${user.email} does not have bigquery.jobs.create permission in project $projectName"
```

Comment:
from @breilly2 via slack: 
> If it’s just this test that cares, it could be pulled higher inside `OrchestrationApiSpec`.
> 
> Another thought though is that checking for the entirety of that message is potentially too brittle.
> 
> I might favor checking for a few smaller pieces of it. A person reading it would be most interested in knowing that it displays the email address, project name, and the permission name, i.e. “bigquery.jobs.create”.
> 
> The email and project name are dynamic so no constants to extract there. The permission name is sufficiently static and meaningful that it doesn’t even really need a constant. If we really wanted a constant for that, it should probably come from a Google client library.

Reply:
(this would apply for each message that is verified)

Reply:
ok